rca/editorial/models.py

```diff
@@ -0,0 +1,116 @@
+from django.db import models
+from modelcluster.fields import ParentalKey
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    PageChooserPanel,
+)
+from wagtail.api import APIField
+from wagtail.core.fields import RichTextField
+from wagtail.images.edit_handlers import ImageChooserPanel
+
+from rca.utils.models import BasePage, RelatedPage
+
+
+class EditorialPageRelatedSchoolsAndResearchPages(RelatedPage):
+    source_page = ParentalKey(
+        "EditorialPage", related_name="related_schools_and_research_pages"
+    )
+    panels = [
+        PageChooserPanel("page", ["schools.SchoolPage", "research.ResearchCentrePage"])
+    ]
+
+    api_fields = [APIField("page")]
+
+
+class Author(models.Model):
+    name = models.CharField(max_length=128)
+
+    def __str__(self):
+        return self.name
+
+
+class EditorialPageAuthor(models.Model):
+    page = ParentalKey("EditorialPage", related_name="editorial_authors")
+    author = models.ForeignKey("Author", related_name="+", on_delete=models.CASCADE)
+    panels = [FieldPanel("author")]
+
+    def __str__(self):
+        return self.author.name
+
+
+class EditorialPage(BasePage):
+
+    intro_text = RichTextField(blank=True, max_length=255)
+    #  this is not on the base page
+    hero_image = models.ForeignKey(
+        "images.CustomImage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    video = models.URLField(blank=True)
+    video_caption = models.CharField(
+        blank=True,
+        max_length=80,
+        help_text="The text displayed next to the video play button",
+    )
+    video_preview_image = models.ForeignKey(
+        "images.CustomImage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    #  do you want it to be from date created - auto_now_add (I think )
+    #  or from date_updated = add_add
+    date = models.DateTimeField(auto_now_add=True)
```

Comment:
I think we want auto_add_now as this adds a date which relates to when the object was first created eg. when the article was first published?

Reply:
:+1: